clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -51,9 +51,13 @@
     private static final String PARTITION_HEADER_KEY_NAME = "partition_header";
     private static final String PARTITION_KEY_NAME = "partition";
     private static final String ERROR_CODE_KEY_NAME = "error_code";
+    private static final String HIGH_WATERMARK_KEY_NAME = "high_watermark";
+    private static final String LOG_START_OFFSET_KEY_NAME = "log_start_offset";
 
     // Default throttle time
     private static final int DEFAULT_THROTTLE_TIME = 0;
+    // Default low watermark
+    private static final long DEFAULT_LOG_START_OFFSET = 0L;
```

Comment:
In general, we want to identify the state of the system as clear as possible. The follower should not take any action if the LOG_START_OFFSET on the broker is NO_LOG_START_OFFSET. But if the follower sees the leader returning the starting offset = 0 while the actual starting offset on the leader is not, this introduces confusion.